codegen/lib/templates/kotlin/android_class.erb

```diff
@@ -60,4 +69,44 @@ actual class <%= entity.name %> private constructor(
 <%- end -%>
     }
 <%  end -%>
+
+<%  unless entity.methods.select{ |x| x.name == "Delete" }.empty? -%>
+    internal class <%= entity.name %>PhantomReference private constructor(
+        referent: <%= entity.name %>,
+        private val nativeHandle: Long,
+    ) :
+        java.lang.ref.PhantomReference<<%= entity.name %>>(referent, queue) {
+        companion object {
+            private val references: MutableSet<<%= entity.name %>PhantomReference> = HashSet()
+            private val queue: java.lang.ref.ReferenceQueue<<%= entity.name %>> = java.lang.ref.ReferenceQueue<<%= entity.name %>>()
+
+            init {
+                Thread {
+                    try {
+                        doDeletes()
```

Comment:
Do we spawn a new thread for each instance of HDWallet, AnyAddress etc?
I saw examples do the same, but it's not optimal. What if we have one `PhantomReferenceRegistry` singleton class that spawns one thread, and `AnyAddress.init()` just register the `nativeHandle` with a corresponding delete function.
So the pseudocode will look like:
```kt
class AnyAddress {
  init {
        PhantomReferenceRegistry.getOrCreate().register(nativeHandle, TWAnyAddressDelete)
    }
}
```

```kt
class PhantomReferenceRegistry {
  init {
    Thread {
      doDeletes()
    }
  }

  // Sends a pair of `nativeHandle` and `TWAnyAddressDelete(nativeHandle)` to the spawned thread.
  private val channel: Channel<Pair<Long, (Long)>>
}
```

Reply:
thanks for suggestion, done